crates/harp/src/parser/parse_data.rs

```diff
@@ -108,11 +113,23 @@ impl ParseData {
 
     pub fn filter_top_level(mut self) -> Self {
         let nodes = std::mem::take(&mut self.nodes);
-        self.nodes = nodes.into_iter().filter(|node| node.parent == 0).collect();
+        self.nodes = nodes
+            .into_iter()
+            .filter(|node| matches!(node.kind, ParseDataKind::Node) && node.parent == 0)
+            .collect();
         self
     }
 }
 
+impl ParseDataNode {
+    pub fn as_point_range(&self) -> std::ops::Range<(usize, usize)> {
+        std::ops::Range {
+            start: (self.line.start, self.column.start),
+            end: (self.line.end, self.column.end),
+        }
+    }
+}
```

Comment:
Not used?

Also feels like it should return a typed `ParseDataRange` to be more self documenting

(I've noticed that the ruff team uses custom types for almost _everything_, and I really think it makes the code easier to read)

Reply:
In general I think it's a balance because too much vocabulary can also make code opaque instead of transparent.

But I'm willing to try and make more type aliases to see how this feels.